include/openthread/platform/radio.h

```diff
@@ -239,24 +239,26 @@ typedef struct otRadioFrame
             uint8_t         mMaxFrameRetries; ///< Maximum number of retries allowed after a transmission failure.
 
             /**
-             * Indicates whether the frame is a retransmission or not.
+             * Indicates whether frame counter, key id, sequence and CSL IEs are properly updated in the header.
              *
              * If the platform layer does not provide `OT_RADIO_CAPS_TRANSMIT_SEC` capability, it can ignore this flag.
              *
              * If the platform provides `OT_RADIO_CAPS_TRANSMIT_SEC` capability, then platform is expected to handle tx
              * security processing and assignment of frame counter. In this case the following behavior is expected:
              *
-             * When `mIsARetx` is set, it indicates that OpenThread core has already set the frame counter and key id
-             * (if security is enabled) in the prepared frame. The counter is ensured to match the counter value from
-             * the previous attempts of the same frame. The platform should not assign or change the frame counter (but
-             * may still need to perform security processing depending on `mIsSecurityProcessed` flag).
+             * When `mIsHeaderUpdated` is set, it indicates that OpenThread core has already set the frame counter and
+             * key id (if security is enabled) in the prepared frame. The counter is ensured to match the counter value
```

Comment:
Mentioning `key id` here is misleading. Radio platform with `OT_RADIO_CAPS_TRANSMIT_SEC` expect `key id` to be set regardless `mIsHeaderUpdated` value. `mIsHeaderUpdated = false` indicates that Frame Counter and CSL IE are not set.